hypothesis-python/src/hypothesis/internal/compat.py

```diff
@@ -125,14 +125,24 @@ def get_type_hints(thing):
 else:
 
     def get_type_hints(thing):
-        if inspect.isclass(thing) and not hasattr(thing, "__signature__"):
-            if is_typed_named_tuple(thing):
-                # Special handling for typing.NamedTuple
-                return thing._field_types  # type: ignore
-            thing = thing.__init__  # type: ignore
+        """Like the typing version, but tries harder and never errors.
```

Comment:
Now _this_ is a docstring. Can we have these sorts of docs from all of `hypothesis.internal` please?

Reply:
:joy: Be the change you want to see in the world!